portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -28,44 +35,75 @@
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    // Get the count from the request.
+    int commentCount = getCommentCount(request);
+
+    // Prepare query and get all comments in Datastore.
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    // Create a list of comments based on results, with a limited number of comments.
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable(Builder.withLimit(commentCount))) {
+      Comment comment = new Comment(
+        (String) entity.getProperty("content"), 
+        (String) entity.getProperty("author"), 
+        (long) entity.getProperty("timestamp"));
+      comments.add(comment);
+    }
+
+    // Converts comments into a JSON string using the Gson library.
+    Gson gson = new Gson();
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(comments));
   }
 
   /**
-   * For a get request, return a JSON version of a quote and person.
+   * For a post request, get all attributes, create an entity, fill the attributes, and store entity.
    */
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    QuotePerson quote = quotes.get((int) (Math.random() * quotes.size()));
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    String content = request.getParameter("comment-content");
+    String author = request.getParameter("comment-author");
+    long timestamp = System.currentTimeMillis();
 
-    String json = convertToJsonUsingGson(quote);
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("timestamp", timestamp);
+    commentEntity.setProperty("content", content);
+    commentEntity.setProperty("author", author);
 
-    response.setContentType("application/json;");
-    response.getWriter().println(json);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/index.html");
   }
 
   /**
-   * Converts a ServerStats instance into a JSON string using the Gson library. Note: Gson library 
-   * dependency added to pom.xml.
+   * Get the number of comments from the request.
    */
-  private String convertToJsonUsingGson(QuotePerson quote) {
-    Gson gson = new Gson();
-    String json = gson.toJson(quote);
-    return json;
+  private int getCommentCount(HttpServletRequest request) {
+
+    // Convert the string version of the count to a float and define max.
+    Float commentCount = Float.parseFloat(request.getParameter("count"));
+    final int MAX_COMMENT_COUNT = 30;
+
+    if (commentCount < 0) {
```

Comment:
Why do you need this case?

Reply:
You can input negative values. It won't break anything, but it will log an error.

Reply:
I am removing this in implementing pagination anyway.